sympy/polys/domains/finitefield.py

```diff
@@ -43,10 +45,11 @@ def _modular_int_factory(mod, dom, symmetric, self):
             flint.nmod(0, mod)
         except OverflowError:
             # Use fmpz_mod
-            ctx = flint.fmpz_mod_ctx(mod)
+            fctx = flint.fmpz_mod_ctx(mod)
+            ctx = lambda x: fctx(operator.index(x))
```

Comment:
I guess this would have to be like
```python
try:
    obj = fctx(x)
except TypeError:
    obj = fctx(index(x))
```
I think it should also get fixed on the python-flint side though to use `index` there so eventually neither change here would be needed.